report/4.Solution/LanuageOfCompiler.tex

```diff
@@ -1,21 +1,29 @@
 \section{The languages of the compiler}
 \label{langsOfCompiler}
-When creating a compiler there are three languages of relevance, as seen in the tombstone diagram \ref{fig:TStoneExample}.
+When creating a compiler there are three languages of relevance, as seen in the tombstone diagram \ref{fig:TStoneOnlyHCL}.
 
 \begin{figure}[H]
 	\centering
-	\includegraphics[width=\textwidth/2+\textwidth/4]{4.Solution/images/T-diagram.png}
+	\begin{tikzpicture}
+	\begin{tombstonediagram}
+	\compiler{cmp}{HCL}{C++}{JVM}{}[scale=2.00]
+	\program[anchor=prg-2-2.north east]{prg}{Source}{HCL}{at (cmp-1-1.north west)}[scale=2.00]
+	\interpreter{int}{JVM}{x86}{at (cmp-2-2.south west)}[scale=2.00]
+	\program{prgo}{Source}{C++}{at (cmp-1-3.north east)}[scale=2.00]
+	\machine{mac}{x86}{at (int-2-1.south west)}[scale=2.00]
+	\end{tombstonediagram}
+	\end{tikzpicture}
 	\caption{
-		Example of Tombstone diagram showcasing an example of three languages for a given compiler\cite{TStoneWiki}.
+		A tombstone diagram, showing the compilation from HCL to C++\cite{TStoneWiki}.
 	}
-	\label{fig:TStoneExample}
+	\label{fig:TStoneOnlyHCL}
 \end{figure}
 In the case of the HCL compiler the three languages are as follows:
 
 \begin{enumerate}
 \item Input: \textbf{HCL} \\
 \item Output: \textbf{Ardunio C++} \\
-\item Compiler: \textbf{Kotlin} \\
+\item Compiler: \textbf{Kotlin, executed on a JVM} \\
```

Comment:
maybe the JVM, but minor minor note, both can work

Reply:
but isnt that a general note regarding Kotlin?